pkg/agent/svid/rotator.go

```diff
@@ -16,23 +17,19 @@ import (
 )
 
 type Rotator interface {
-	Start()
-	Stop()
+	Run(ctx context.Context) error
 
 	State() State
 	Subscribe() observer.Stream
 }
 
 type rotator struct {
 	c      *RotatorConfig
-	stop   chan struct{}
-	done   chan struct{}
 	client client.Client
 
 	state observer.Property
 
-	m       sync.RWMutex
-	running bool
+	m sync.RWMutex
```

Comment:
Yep, no longer used. I'll remove. Regarding it not being a pointer, thats technically fine as long as you aren't passing the mutex around. In this case it was safe because only struct functions were using it.

Reply:
I'm surprised - the docs + source show all the exported methods as being defined with a pointer receiver: https://golang.org/pkg/sync/#RWMutex

Reply:
Methods with a pointer receiver don't need to be called via a pointer variable. If called with a non-pointer value, the method receives a pointer to that value.

https://play.golang.org/p/muIovpqceAS

Reply:
Huh, TIL...

I dug through the language spec looking for more information about this. The [method set](https://golang.org/ref/spec#Method_sets) section makes it seem like this should not be the case, but referencing the [calls section](https://golang.org/ref/spec#Calls) turns up the answer:

`
A method call x.m() is valid if the method set of (the type of) x contains m and the argument list can be assigned to the parameter list of m. If x is addressable and &x's method set contains m, x.m() is shorthand for (&x).m()
`

Reply:
nice! i spent a minute trying to find it and couldn't.